sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,240 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division, absolute_import
+
+import numbers
+import numpy as np
+import warnings
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.preprocessing.data import _transform_selected
+from sklearn.utils.validation import (
+    check_array,
+    check_is_fitted,
+    column_or_1d,
+    FLOAT_DTYPES
+)
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int or array-like, shape (n_features,) (default=2)
+        The number of bins to produce. The intervals for the bins are
+        determined by the minimum and maximum of the input data.
+        Raises ValueError if ``n_bins < 2``.
+
+        If ``n_bins`` is an array, and there is a ignored feature at
+        index ``i``, ``n_bins[i]`` will be ignored.
+
+    ignored_features : int array-like (default=None)
+        Column indices of ignored features. (Example: Categorical features.)
+        If ``None``, all features will be discretized.
```

Comment:
I am fine with the `If None ...`. What I don't find enough explicit is `(Examples: Categorical features.)`

```
Column indices of ignored features, e.g. categorical features for
which this transformer is not suitable. If ``None``, all features will
be discretized.
```